backend/compact-connect/lambdas/purchases/handlers/utils.py

```diff
@@ -109,3 +109,36 @@ def caught_handler(event, context: LambdaContext):
             raise
 
     return caught_handler
+
+
+class authorize_compact:  # noqa: N801 invalid-name
+    """Authorize endpoint by matching path parameter compact to the expected scope, (i.e. aslp/read)"""
```

Comment:
So, we actually need to do a slightly different authz decorator here because it isn't done exactly the way you need elsewhere. What this decorator should be looking for is `f'{compact}/{compact}.admin'`, which is close to but not quite what `authorize_compact_jurisdiction` does in `provider-data-v1`.